### PR TITLE
External ingress

### PR DESCRIPTION
**Pull request by chrisstafford**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202594685849644

### CODE REVIEW

**Comment by chrisstafford** [comment]
I still need to create the secrets for the certificate and still need clarity on whether or not we want an API endpoint exposed outside the network. I would however like a sanity check that this at least looks correct.

File: cluster-scope/overlays/nerc-ocp-prod/clustersecretstores/nerc-cluster-secrets_patch.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: external-secrets.io/v1beta1
```
Comment (larsks) [inline cluster-scope/overlays/nerc-ocp-prod/clustersecretstores/nerc-cluster-secrets_patch.yaml:1]: For the production cluster,  we're going to want per-namespace secretstores. A cluster-wide secret store would potentially expose sensitive secrets to all cluster users.

File: cluster-scope/overlays/nerc-ocp-prod/ingresscontrollers/external-apps.ingress-controller.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+apiVersion: operator.openshift.io/v1
```
Comment (larsks) [inline cluster-scope/overlays/nerc-ocp-prod/ingresscontrollers/external-apps.ingress-controller.yaml:1]: This ingress controller needs to be configured to listen on a public address. That means we're first going to need to configure an appropriate interface on all the worker nodes (so probably a `NodeNetworkConfigurationPolicy` resource), and then we're going to need to perform some work -- of which I don't know the specifics -- to bind the ingress controller to the public address.

**Review by larsks — CHANGES_REQUESTED** [review]
I've requested one specific change, but I'd like to make a couple more broad suggestions.

I prefer to see pull requests on a particular feature (e.g., this one would be "Configure external ingress"). Things that are dependencies of that feature, but not directly related (like configuring secret stores, or configuring the API certificate) should be split out into separate pull requests. This has a bunch of advantages:

- Smaller pull requests are easier to review
- It allows us to merge the dependencies while still working on other parts of the PR
- It may enable other folks to work on unrelated pull requests that share the same dependencies

**Comment by chrisstafford** [comment]
@larsks, I agree on all points, I just branched this from a branch still waiting to be merged. I'll close this PR and create a new branch from the main and include your changes.

**Comment by naved001** [comment]
@chrisstafford should this be closed?

**Comment by joachimweyl** [comment]
@chrisstafford ^

**Pull request closed without merging**